authors/apps/articles/views.py

```diff
@@ -85,3 +86,55 @@ def destroy(self, request, pk=None):
         article.delete()
         return Response({"message": "article deleted successfully"},
                         status=status.HTTP_200_OK)
+
+
+class LikeViewSet(viewsets.ViewSet):
+    serializer_class = ArticleSerializer
+    permissions = (IsAuthenticatedOrReadOnly,)
+
+    def partial_update(self, request, pk=None):
+        """Update likes field."""
+        try:
+            article = Article.objects.get(pk=pk)
+        except Exception:
+            raise NotFound("The article does not exist")
+        if article in Article.objects.filter(
```

Comment:
It'll be faster to just check if the user is in the article.disliked_by queryset.

Reply:
ok...working on it